### PR TITLE
The run report shows the difference it had already worked out

### LINKED ISSUE #263: The golden run report says a failure did not reproduce the answer key when it did, and hides the claim that explains why
Four defects in the golden run report, all in the renderer and its template. Every one of them is
a presentation problem over data the run already computed and persisted, so none needs a change to
the comparator, the claim reader or the contract.

All four show up on one very ordinary failure: an item whose rows matched the answer key exactly and
which failed a `must_filter` gate. The page renders that as:

> Did not reproduce the answer key · accuracy 1.000 · the dataset requires a filter this statement
> does not write

A reader gets a contradiction, a number that appears to disprove the sentence beside it, and no way
to find out what actually went wrong short of diffing two SQL statements by eye.

## 1. The verdict sentence is factually wrong on a gated item

`shared/golden-run-template.html`:

```js
const verdict = item.passed ? "Reproduced the answer key" : "Did not reproduce the answer key";
```

`passed` is false whenever a gate fires, including when the result sets are identical. On the item
above the generated statement reproduced the answer key perfectly — 6 of 6 rows, accuracy 1.0 — and
the page says it did not.

The verdict should be read from the **score**, not from the pass flag. Reproduction and passing are
different questions, which is the whole reason structure gates separately from rows. Something like
"Reproduced the answer key, but did not answer the question asked" carries both facts.

## 2. `accuracy 1.000` is shown beside a failure

`render_golden_run.py` already anticipates this confusion and guards half of it:

```python
# The largest accuracy that may be shown for an item that did not score exactly 1.0. Rounding alone
# ... beside the sentence saying it did not reproduce the answer key. That is the one confusion this
```

The cap stops a near-miss (0.9996) being *rounded* to 1.000 next to a failure. But it never
considered the case where the accuracy is genuinely 1.0 and the item failed a gate — which is not an
edge case, it is the normal shape of a `must_filter` catch. The guard covers rounding and misses
gating.

## 3. The gate sentence names no column, and assumes there is only one kind of gate

```js
text: " · the dataset requires a filter this statement does not write",
```

A hardcoded string. The item's own payload carries what the reader needs:

```json
"gates": [{
  "kind": "must_filter",
  "column": "opened",
  "reason": "the dataset requires this column to be filtered, and the generated statement
             constrains it in none of the predicates it writes"
}]
```

The column name is right there and is not printed, so the page states that *a* filter is missing and
leaves the reader to work out *which* by reading two statements.

It is also wrong for the other gate. The contract says structure gates exactly twice — on
`must_filter` **and** on a differing resolved date window — but `item.gated` renders this
filter-specific sentence either way, so a date-window failure is described as a missing filter.

## 4. The page keeps the one claim that agrees and drops the one that explains the failure

This is the substantive one.

```python
def _tables_claim(item: dict) -> Optional[dict[str, Any]]:
    ...
    claim = claims[0]
```

```js
function deltaLine(item) {
  const claim = item.tables;
```

The projection keeps `claims[0]` — `tables` — and discards the other six. In a structural failure
`tables` almost always **agrees**: both statements read the same table, which is why the comparison
got far enough to gate on something else. So the page reliably shows the claim that says nothing.

Meanwhile the claim that says everything is computed, written to the JSON artifact beside the page,
and never rendered:

```json
{
  "name": "filter_predicates",
  "status": "differs",
  "generated": ["gte(created, '2024-01-01')", "lt(created, '2025-01-01')"],
  "golden":    ["gte(opened, '2024-01-01')", "lt(opened, '2025-01-01')"]
}
```

That is the entire diagnosis — the statement filtered on the record's audit timestamp instead of the
business event timestamp — in a form a person can read in two seconds without looking at any SQL.

Rendering the claims whose status is `differs` would turn the report from "here are two statements,
spot the difference" into "here is the difference". The claim reader exists to produce exactly this,
and F22 §3 says the rest of the claim difference "is rendered for a person" — currently it isn't.

## Why this matters more than it looks

The terminal deliberately prints no SQL, so the report is the **only** surface where a failure can be
understood. A run of fifteen items where six failed the same way took several minutes to diagnose by
hand, from a page that had already worked out the answer and kept it in a file beside itself.

## Scope

`plugins/agami/scripts/render_golden_run.py` (widen the projection past `claims[0]`, and reconsider
`_shown` for a gated 1.0) and `plugins/agami/shared/golden-run-template.html` (verdict sentence, gate
sentence, render the differing claims). No comparator, claim-reader or contract change. The run's
JSON already carries every value needed.

Found on agami-core `main` (0.8.0) while running a first real dataset against a live warehouse.

### PR DESCRIPTION
**Pull request by ashwin-agami**
Closes #263.

Found by reading the page against a real fifteen-case run. Every change is presentation over values the run already computed — the comparator, the claim reader and the contract are untouched.

The failure that motivated all of it rendered like this:

> Did not reproduce the answer key · accuracy 1.000 · the dataset requires a filter this statement does not write

A contradiction, a number that appears to disprove the sentence beside it, and no way to find out what actually went wrong short of diffing two SQL statements by eye.

## 1. The verdict was false on a gated item

`const verdict = item.passed ? … : "Did not reproduce the answer key"` — but `passed` goes false whenever a gate fires, including when the result sets are identical. That item reproduced its answer key perfectly, 6 of 6 rows.

Reproducing and passing are different questions — which is the whole reason structure gates separately from rows — so both now travel and the page says both: *"Reproduced the answer key, but did not answer the question asked."*

## 2. The accuracy leaves the page

Across a real fifteen-case run **every accuracy was exactly 1.0 or exactly 0.0**. That is structural: a row-count difference, an unpaired column and an extra column each short-circuit before an overlap is computed, and `match_columns` pairs only on equal value vectors, so a column that is wrong anywhere does not pair at all. The one route to a value in between — every column carrying the right set of values while the rows do not line up — is real and rare.

And on exactly that occasion `_score_values` already writes the same fact in words: *"5 of the answer key's 15 rows matched."* So the number was redundant when it meant something and misleading the rest of the time.

This also dissolves the `1.000`-beside-a-failure defect rather than patching it. The existing guard capped a *rounded* near-miss at 0.999; nobody had considered a genuine 1.0 next to a gate, which is the normal shape of a `must_filter` catch.

The score is unchanged where it is computed and still decides the verdict.

> Worth recording for later: what a reader actually wants at 5-of-15 is *which ten rows differed*, and the report cannot show that — rows never reach a run result by design. The ratio was a proxy for something deliberately withheld. Changing that is a contract conversation about the no-rows rule, not a renderer change.

## 3. A gate names its column

The page printed one hardcoded string that named no column, so a reader was told *a* filter was missing and left to work out which. It was also written as though `must_filter` were the only gate; structure gates twice, and a differing date window rendered as a missing filter.

## 4. The page kept the claim that agrees and dropped the one that explains

The substantive one. The projection kept `claims[0]` — `tables` — and discarded the other six.

In a structural failure `tables` almost always **agrees**: both statements read the same table, which is why the comparison got far enough to gate on something else. So the page reliably showed the claim that says nothing, while this was computed, written to the JSON artifact beside the page, and never rendered:

```
filter_predicates  DIFFERS
  generated:  gte(created, '2024-01-01'), lt(created, '2025-01-01')
  answer key: gte(opened, '2024-01-01'), lt(opened, '2025-01-01')
```

That is the whole diagnosis, readable without looking at any SQL.

Two shapes needed handling on the way, and both came from the real run rather than from imagination: a resolved date window is an **object**, not a list, and the page called `Array.join` on these — so an object would have thrown and left the report blank. It is flattened to `created in [2024-01-01, 2025-01-01)`, half-open because that is the point of resolving one. And `ordering` writes **pairs** (`[["incident_count", "desc"]]`), which the old coercion dropped as "nested", rendering the claim that most often differs as `generated none · answer key none`.

## 5 and 6. Timestamp and tiles

`2026-09-06T11:47:38+00:00` is a sortable key; the page has no second place where it is a key. It now reads `6 September 2026 at 11:47 UTC`.

The header always drew five tiles, three of them zero on a healthy run. Now: Cases, Passed, Failed always; "Not compared" and "Couldn't run" only when non-zero, with a sentence when they appear.

**They are deliberately not merged into "failed."** This project's own first run came back `3 errored, 0 failed`, which under a three-bucket summary reads as green when in fact the generator was broken and nothing had run — exactly the distinction the `1` / `2` exit codes exist to make. Same for unscored: both sides returning no rows is not agreement.

## Plus: filter by outcome

A corpus-sized run is hundreds of cases and was a flat scroll. Display only — sections are hidden, nothing is recounted or re-sorted, so the tiles and section headings keep describing the whole run. The control does not appear when there is nothing to choose between.

## Tests

`tests/test_render_golden_run.py` goes from 37 to 40. Several existing tests pinned the old behaviour and were rewritten rather than deleted — including `test_a_gated_item_shows_a_perfect_score_beside_a_failing_verdict`, which asserted the confusing output as intended, and the two accuracy-rounding tests whose subject no longer reaches the page.

Full `dev.py check` green: 5464 passed, 12 skipped.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: plugins/agami/scripts/render_golden_run.py [hunk 1]
```diff
@@ -68,59 +73,140 @@
...
-    the number beside a pass.
+    It read `2026-09-06T11:47:38+00:00`, which is a sortable key rather than a sentence, and the
+    page has no second place where the timestamp is a key. UTC is spelled out because the reader is
+    not necessarily in it, and the day is not zero-padded because nobody says "06 September".
     """
-    if accuracy >= 1.0:
-        return round(accuracy, _ACCURACY_DECIMALS)
-    return min(round(accuracy, _ACCURACY_DECIMALS), _NEARLY_ONE)
+    stamp = now or datetime.datetime.now(datetime.timezone.utc)
+    return f"{stamp.day} {stamp:%B %Y} at {stamp:%H:%M} UTC"
```
[the remark this answers is not in the document]
Comment (ashwin-agami) [reply plugins/agami/scripts/render_golden_run.py]: Fixed — month names come from an explicit table now. Worth adding that the consequence was worse than variance: the test for this asserts a fixed string, so it would have passed on the author's machine and failed on a contributor's.

File: plugins/agami/scripts/render_golden_run.py [hunk 2]
```diff
@@ -68,59 +73,140 @@
...
+    rather than described. This is one of the two claims that can gate, which is why its shape is
+    handled here instead of falling through to the generic rendering below.
+    """
+    column = value.get("column")
+    start, end = value.get("start"), value.get("end")
+    if not column or (start is None and end is None):
+        return ""
+    open_bracket = "[" if value.get("start_inclusive", True) else "("
+    close_bracket = "]" if value.get("end_inclusive", False) else ")"
+    return f"{column} in {open_bracket}{start}, {end}{close_bracket}"
```
[the remark this answers is not in the document]
Comment (ashwin-agami) [reply plugins/agami/scripts/render_golden_run.py:299]: Fixed, and this was the most likely of the four to be hit — a single lower bound (`placed_at >= '2024-01-01'`) is an ordinary predicate, not a corner case, and it resolves to a start with no end. Open bounds now render as a comparison rather than an interval: `placed_at >= 2024-01-01`. Covered by a test asserting the word `None` appears in neither side.

File: plugins/agami/scripts/render_golden_run.py [hunk 3]
```diff
@@ -68,59 +73,140 @@
...
+    """
+    block = item.get("claims")
+    gates = block.get("gates") if isinstance(block, dict) else None
+    if not isinstance(gates, list):
+        return []
+    return [
+        {"kind": str(gate.get("kind", "")), "column": str(gate.get("column") or "")}
+        for gate in gates
+        if isinstance(gate, dict)
+    ]
```
[the remark this answers is not in the document]
Comment (ashwin-agami) [reply plugins/agami/scripts/render_golden_run.py:389]: Fixed — `gate.get("kind") or ""`, and a gate with no kind is dropped rather than rendered. The default in `.get(key, "")` never fires for a key that is present and null, which is exactly the shape a hand-edited artifact produces.

File: plugins/agami/shared/golden-run-template.html [hunk 4]
```diff
@@ -223,46 +249,52 @@ <h1 class="dashboard-title">{{REPORT_TITLE}}</h1>
...
+    // in the JSON beside this page, and never rendered.
+    function deltaLines(item) {
+      const claims = item.claims || [];
+      if (!claims.length) {
+        return [el("div", {
           class: "delta",
-          text: "tables: no statement was read, so there was nothing to compare",
-        });
+          text: "no statement was read, so there was nothing to compare",
+        })];
```
[the remark this answers is not in the document]
Comment (ashwin-agami) [reply plugins/agami/shared/golden-run-template.html:331]: Fixed. The two situations are now told apart on `generated_sql`: a case that produced no statement says so, and a case that has both statements but no claims says they were not compared and sends the reader to them. Agreed that the old message was misleading in precisely the error-tolerance case the projection exists for.

**Comment by ashwin-agami** [comment]
@copilot review — this has changed substantially since your last pass. A review panel found several things after it, and the fixes are pushed: a missing relevance floor that would have stopped every save, a report line that was false on two shapes, an environment-allowlist test that could not fail, and the empty-library case that wrote no test at all. Please re-review the current state rather than the diff you saw.

File: plugins/agami/scripts/render_golden_run.py [hunk 5]
```diff
@@ -67,60 +72,315 @@
...
+
+    It read `2026-09-06T11:47:38+00:00`, which is a sortable key rather than a sentence, and the
+    page has no second place where the timestamp is a key. UTC is spelled out because the reader is
+    not necessarily in it, and the day is not zero-padded because nobody says "06 September".
     """
-    if accuracy >= 1.0:
-        return round(accuracy, _ACCURACY_DECIMALS)
-    return min(round(accuracy, _ACCURACY_DECIMALS), _NEARLY_ONE)
+    stamp = now or datetime.datetime.now(datetime.timezone.utc)
+    return f"{stamp.day} {_MONTHS[stamp.month - 1]} {stamp.year} at {stamp:%H:%M} UTC"
```
[the remark this answers is not in the document]
Comment (ashwin-agami) [reply plugins/agami/scripts/render_golden_run.py:110]: Both fixed on the branch. Worth noting the unknown-status guard sits AFTER the `error` and `unscored` branches, so a legitimately unscored item still gets its own sentence rather than the catch-all.

File: plugins/agami/shared/golden-run-template.html [hunk 6]
```diff
@@ -223,46 +281,87 @@ <h1 class="dashboard-title">{{REPORT_TITLE}}</h1>
         line.appendChild(el("span", { text: "Nothing could be compared — " + item.reason }));
         return line;
       }
-      const verdict = item.passed ? "Reproduced the answer key" : "Did not reproduce the answer key";
-      line.appendChild(el("span", { text: verdict + " · accuracy " }));
-      line.appendChild(el("span", { class: "score", text: score(item.accuracy) }));
+      // Built from what the comparison decided and what the gates decided, never from `passed`
```
[the remark this answers is not in the document]
Comment (ashwin-agami) [reply plugins/agami/shared/golden-run-template.html]: Both fixed on the branch. Worth noting the unknown-status guard sits AFTER the `error` and `unscored` branches, so a legitimately unscored item still gets its own sentence rather than the catch-all.

**Comment by ashwin-agami** [comment]
recheck
